Model not seting default values after building a factory

Creating a model after building a factory of the model rises a validation error of the default values not being defined.

Example:
```python
class TestModel(ormar.Model):
    class Meta(ormar.ModelMeta):
        database = database
        metadata = meta

    id: UUID4 = ormar.UUID(primary_key=True, default=uuid4)
    text: str = ormar.Text()
    created_date: datetime.datetime = ormar.DateTime(default=datetime.datetime.now)


class TestModelFactory(OrmarModelFactory):
    __model__ = TestModel


def test_test_model():
    TestModelFactory.build()
    TestModel(text='qwerty')
```
results in:
```
>               raise validation_error
E               pydantic.error_wrappers.ValidationError: 2 validation errors for TestModel
E               id
E                 field required (type=value_error.missing)
E               created_date
E                 field required (type=value_error.missing)
```

Removing TestModelFactory.build() works as intended.


The problem persists if the factory is called from one test and the instancing in the model in another one.
Ex.
```python
def test_test_model_build():
    TestModelFactory.build()

def test_test_model():
    TestModel(text='qwerty')
```

@jtraub Are you inheriting TestModel from ormar.Model or BaseModel?
When inheriting from BaseModel i get the same error than you.
When inheriting form ormar.Model i get:
```
>               raise validation_error
E               pydantic.error_wrappers.ValidationError: 1 validation error for TestModel
E               state
E                 field required (type=value_error.missing)
```

When not building the Factory, it runs as expected.
I should also add that i'm running python 3.11 (FROM python:3.11.0-slim-bullseye) but i think you already know that.

@jtraub yes, that fixes the enum problem, thanks!

Now i have found a problem with nullable fields, sorry for not caching it beforehand
To replicate set text to:
text: str = ormar.Text(nullable=True)
and 
DemoTestModel()
you will get:
E               pydantic.error_wrappers.ValidationError: 1 validation error for DemoTestModel
E               text
E                 field required (type=value_error.missing)
(https://collerek.github.io/ormar/fields/common-parameters/#nullable)

I also found some tests that are failing with the presence of other tests (might not be related, i have to look into it more)

--- a/pydantic_factories/extensions/ormar_orm.py
+++ b/pydantic_factories/extensions/ormar_orm.py
@@ -2,6 +2,7 @@ import random
 from typing import TYPE_CHECKING, Any, Union
 
 from pydantic import BaseModel
+from pydantic.utils import smart_deepcopy
 
 from pydantic_factories.factory import ModelFactory
 from pydantic_factories.utils import is_pydantic_model, is_union
@@ -20,7 +21,10 @@ class OrmarModelFactory(ModelFactory[Model]):  # pragma: no cover # type: ignore
     def get_field_value(cls, model_field: "ModelField", field_parameters: Union[dict, list, None] = None) -> Any:
         """We need to handle here both choices and the fact that ormar sets values to be optional."""
 
-        model_field.required = True
+        if not model_field.required:
+            model_field = smart_deepcopy(model_field)
+            model_field.required = True
+
         # check if this is a RelationShip field
         if (
             is_union(model_field=model_field)

--- a/tests/extensions/test_ormar_extension.py
+++ b/tests/extensions/test_ormar_extension.py
@@ -1,12 +1,16 @@
+from datetime import datetime
 from enum import Enum
-from typing import TYPE_CHECKING
+from uuid import uuid4
 
 import pytest
+from pydantic import UUID4
 
 try:
     import sqlalchemy
     from databases import Database
-    from ormar import DateTime, ForeignKey, Integer, Model, String
+    from ormar import UUID, DateTime
+    from ormar import Enum as OrmarEnum
+    from ormar import ForeignKey, Integer, Model, String, Text
     from sqlalchemy import func
 
     from pydantic_factories.extensions import OrmarModelFactory
@@ -17,10 +21,6 @@ postgres_dsn = "postgresql+asyncpg://pydantic-factories:pydantic-factories@postg
 
 database = Database(url=postgres_dsn, force_rollback=True)
 metadata = sqlalchemy.MetaData()
-
-
-if TYPE_CHECKING:
-    from datetime import datetime
 
 
 class BaseMeta:
@@ -35,8 +35,8 @@ class Mood(str, Enum):
 
 class Person(Model):
     id: int = Integer(autoincrement=True, primary_key=True)
-    created_at: "datetime" = DateTime(timezone=True, server_default=func.now())
-    updated_at: "datetime" = DateTime(timezone=True, server_default=func.now(), onupdate=func.now())
+    created_at: datetime = DateTime(timezone=True, server_default=func.now())
+    updated_at: datetime = DateTime(timezone=True, server_default=func.now(), onupdate=func.now())
     mood: Mood = String(choices=Mood, max_length=20)  # type: ignore
 
     class Meta(BaseMeta):
@@ -75,4 +75,23 @@ def test_job_factory() -> None:
 
     assert result.id
     assert result.name == job_name
-    assert result.person
+    assert result.person is not None
+
+
+def test_model_creation_after_factory_build() -> None:
+    # https://github.com/starlite-api/pydantic-factories/issues/128
+    class TestModel(Model):
+        class Meta(BaseMeta):
+            pass
+
+        id: UUID4 = UUID(primary_key=True, default=uuid4)
+        text: str = Text()
+        text2: str = Text(nullable=True)
+        created_date: datetime = DateTime(default=datetime.now)
+        mood: Mood = OrmarEnum(enum_class=Mood, default=Mood.HAPPY)
+
+    class TestModelFactory(OrmarModelFactory):
+        __model__ = TestModel
+
+    TestModelFactory.build()
+    TestModel(text="qwerty")

